metadata/metadata.go

```diff
@@ -286,6 +261,21 @@ func getMetadata(ctx context.Context, hang bool) (*metadata, error) {
 	if err != nil {
 		return nil, err
 	}
-	var ret metadata
+	var ret Descriptor
 	return &ret, json.Unmarshal(md, &ret)
 }
+
+// WriteGuestAttributes does a put call to mds changing a guest attribute value.
+func WriteGuestAttributes(key, value string) error {
+	logger.Debugf("write guest attribute %q", key)
+	client := &http.Client{Timeout: defaultTimeout}
+	finalURL := metadataURL + "instance/guest-attributes/" + key
+	req, err := http.NewRequest("PUT", finalURL, strings.NewReader(value))
```

Comment:
this would create new context on each new http request, why not create it top of the call chain and accept it here as an argument?

Reply:
This function was just moved from one place to another without an actual rewrite or improvement. It will certainly have a rewrite anytime soon.